docs/release-notes/flink-1.12.md

```diff
@@ -0,0 +1,169 @@
+---
+title: "Release Notes - Flink 1.12"
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+These release notes discuss important aspects, such as configuration, behavior,
+or dependencies, that changed between Flink 1.11 and Flink 1.12. Please read
+these notes carefully if you are planning to upgrade your Flink version to 1.12.
+
+* This will be replaced by the TOC
+{:toc}
+
+
+### APIs
+
+#### Remove deprecated methods in ExecutionConfig [FLINK-19084](https://issues.apache.org/jira/browse/FLINK-19084)
+
+Deprecated method `ExecutionConfig#isLatencyTrackingEnabled` was removed, you can use `ExecutionConfig#getLatencyTrackingInterval` instead. 
+
+Deprecated and methods without effect were removed: `ExecutionConfig#enable/disableSysoutLogging`, `ExecutionConfig#set/isFailTaskOnCheckpointError`.
+
+Removed `-q` flag from cli. The option had no effect. 
+
+#### Remove deprecated RuntimeContext#getAllAccumulators [FLINK-19032](https://issues.apache.org/jira/browse/FLINK-19032)
+
+The deprecated method `RuntimeContext#getAllAccumulators` was removed. Please use `RuntimeContext#getAccumulator` instead. 
+
+#### Deprecated CheckpointConfig#setPreferCheckpointForRecovery due to risk of data loss [FLINK-20441](https://issues.apache.org/jira/browse/FLINK-20441)
+
+The `CheckpointConfig#setPreferCheckpointForRecovery` method has been deprecated, because using checkpoints for recovery can lead to data loss.
```

Comment:
```suggestion
The `CheckpointConfig#setPreferCheckpointForRecovery` method has been deprecated, because preferring  older checkpoints over newer savepoints for recovery can lead to data loss.
```